### PR TITLE
Refactor Write API request processing

### PR DESCRIPTION
**Pull request by GRyall**
Refactor the getRequestContent function:
* sort out indentation
* simplify some of the logic

### CODE REVIEW

**Comment by GRyall** [comment]
I've just done a little more refactoring to simplify the logic some more. It is easier to read as two commits, so I have left it that way

**Comment by GRyall** [comment]
Final reworking of all the commits into something reviewable

**Comment by tofu-rocketry** [comment]
What was the problem with that reworked logic, and how did you fix it?

**Comment by GRyall** [comment]
> 
> 
> What was the problem with that reworked logic, and how did you fix it?

I missed a valid case for a variable to be null when reworking the function to be less complex. I have now accounted for that case.

File: htdocs/PI/write/PIWriteRequest.php [hunk 1]
```diff
@@ -179,64 +179,78 @@ private function getAndProcessURL($method, $requestUrl) {
...
-            "The JSON message is not correctly formatted. " . $genericError
+            "A value for \"$this->entityProperty\" should be provided. " .
+            "This should be provided in a JSON string {\"value\":\"<value for " .
+            "\"$this->entityProperty\">\"}, with no other pairs present." . $genericError
           );
         }
-    }
+
+        //If there are additional entiries in our array to the value, throw exception
+        if(count($requestArray)==1) {
```
Comment (gregcorbett) [inline htdocs/PI/write/PIWriteRequest.php]: Seems this should not be `if(count($requestArray)==1)`, greater than 1? not equal to one?

Comment (GRyall) [reply htdocs/PI/write/PIWriteRequest.php]: yes

Comment (GRyall) [reply htdocs/PI/write/PIWriteRequest.php]: resolved in latest commit

**Pull request merged**